x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/create/helpers.ts

```diff
@@ -165,13 +169,16 @@ export const formatAboutStepData = (aboutStepData: AboutStepRule): AboutStepRule
   const resp = {
     author: author.filter((item) => !isEmpty(item)),
     ...(isBuildingBlock ? { building_block_type: 'default' } : {}),
-    ...(isAssociatedToEndpointList
-      ? {
-          exceptions_list: [
-            { id: ENDPOINT_LIST_ID, namespace_type: 'agnostic', type: 'endpoint' },
-          ] as AboutStepRuleJson['exceptions_list'],
-        }
-      : {}),
+    exceptions_list: (isAssociatedToEndpointList
+      ? [
+          { id: ENDPOINT_LIST_ID, namespace_type: 'agnostic', type: 'endpoint' },
+          ...(exceptionsList != null ? exceptionsList : []).filter(
+            (list) => list.type !== 'endpoint'
+          ),
+        ]
+      : exceptionsList != null
+      ? exceptionsList.filter((list) => list.type !== 'endpoint')
+      : []) as AboutStepRuleJson['exceptions_list'],
```

Comment:
I think you could pull `exceptionsList.filter((list) => list.type !== 'endpoint')` out into a `const`. So something like:
```javascript
  const detectionExceptionLists =
    exceptionsList != null ? exceptionsList.filter((list) => list.type !== 'endpoint') : [];

  const resp = {
    ...(isAssociatedToEndpointList
      ? {
          exceptions_list: [
            { id: ENDPOINT_LIST_ID, namespace_type: 'agnostic', type: 'endpoint' },
            ...detectionExceptionLists,
          ] as AboutStepRuleJson['exceptions_list'],
        }
      : exceptionsList != null
      ? {
          exceptions_list: [...detectionExceptionLists],
        }
      : {}),
  };
```
The main difference with this approach is that if `isAssociatedToEndpointList` is `false` AND `exceptions_list` doesn't exist in the rule, you will not be adding a new `exceptions_list` field to the rule. I don't know if we want to set it to an empty `[]` if the field is not there.

Reply:
Yeah, I changed that from the previous version to make it cleaner, and figured the empty list wouldn't be a problem. But your approach is better, I'll switch it!